src/simple.jl

```diff
@@ -97,8 +97,13 @@ end
 return: a solution matrix with time steps in `t` along rows and
         state variable `y` along columns
 """
-function cvode(f::Function, y0::Vector{Float64}, t::Vector{Float64}, userdata::Any=nothing;
-              integrator=:BDF, reltol::Float64=1e-3, abstol::Float64=1e-6)
+cvode(f::Function, y0::Vector{Float64}, t::AbstractVector, userdata::Any=nothing;
+                integrator=:BDF, reltol::Float64=1e-3, abstol::Float64=1e-6) = 
+    cvode!(f, y0, t, zeros(length(t), length(y0)), userdata; 
+                    integrator=integrator, reltol=reltol, abstol=abstol)
+
+function cvode!(f::Function, y0::Vector{Float64}, t::AbstractVector, y::Matrix{Float64}, userdata::Any=nothing;
```

Comment:
By convention, !-functions modify their 1st argument, so I would recommend putting `y` forward.
Also, the size of `y` has to be checked and `DimensionMismatch` thrown if it doesn't meet expectation.
With this check done, you can annotate the `for k` loop with `@inbounds`.